lnst/Recipes/ENRT/TrafficControlRecipe.py

```diff
@@ -0,0 +1,120 @@
+        return False
+
+    def init_traffic_control(self):
+        host = self.matched.host1
+        host.tc_tester: TrafficControlTester = host.init_class(TrafficControlTester, host.eth0.name)
+
+
+    def vf_setup(self):
+        host1: Namespace = self.matched.host1
+        # Create virtual func
+        host1.run(
+            f"echo 1 > /sys/bus/pci/devices/{host1.eth0.bus_info}/sriov_numvfs "
+        )
+
+        #Unbind the interfaces they create
+        # Need to see if there is a better way to do this.
+        vf_pci = host1.run(f"readlink /sys/bus/pci/devices/{host1.eth0.bus_info}/virtfn0")
+        vf_pci = pathlib.PurePath(vf_pci.stdout.rstrip()).name
+        host1.run(f"echo {vf_pci} > /sys/bus/pci/devices/{host1.eth0.bus_info}/virtfn0/driver/unbind")
+
+    @contextmanager
+    def _test_wide_context(self):
+        config = self.test_wide_configuration() # Do I have a use for this? Probably not
+        #self.describe_test_wide_configuration(config)
+        try:
+            yield config
+        finally:
+            self.test_wide_deconfiguration(config)
+
+
+    def test_wide_configuration(self):
+        config = TcTestConfiguration()
+        host = self.matched.host1
+        self.init_traffic_control()
+
+        # Enable steering if its needed
+        if self.is_steering_test():
+            host.run(f"devlink dev param set pci/{host.eth0.bus_info} "
+                     f"name flow_steering_mode value \"{self.params.steering_mode}\" cmode runtime")
+
+        self.vf_setup()
+
+        # Switch to switchdev mode
+        host.run(f"devlink dev eswitch set pci/{host.eth0.bus_info} mode switchdev")
+
+
+        # init ingress qdisc
+        #  This might fail, we might not need it because lnst deletes it on device teardown
+        host.run(f"tc qdisc del dev {host.eth0.name} ingress | true")
+        host.run(f"tc qdisc add dev {host.eth0.name} ingress")
+
+        return config
+
+
+    def test_wide_deconfiguration(self, config):
+        """
+        Cleanup:
+        tc filter del dev ens5f0np0 ingress
+        devlink dev eswitch set pci/$pci mode legacy
+        echo 0 > /sys/bus/pci/devices/$pci/sriov_numvfs
+        """
+        host: Namespace = self.matched.host1
+
+        host.run(f"tc filter del dev {host.eth0.name} ingress") # Probably not needed, it seems to be done by lnst device class
+        host.run(f"echo 0 > /sys/class/net/{host.eth0.name}/device/sriov_numvfs")
+        time.sleep(2)
+        host.run(f"devlink dev eswitch set pci/{host.eth0.bus_info} mode legacy")
+        time.sleep(3)
+
+    def do_tc_test(self):
+        host: Namespace = self.matched.host1
+
+        success, time_taken, messages = host.tc_tester.run_test(self.params.num_rules)
+
+        res = ResultType.PASS if success else ResultType.FAIL
+
+        self.add_result(res, description=" ".join(messages), data={"time_taken": time_taken})
+
+    def test(self):
```

Comment:
Fixed, base it on the order of methods in `BaseEnrtRecipe.py`. Let me know if this order makes sense to you, because as I got down to the more specific methods I wasn't sure whether to go depth first or breath first in the ordering. `BaseEnrtRecipe` seems to be breath first,

Reply:
Did a check of this in the current version and reordered it a bit, let me know if this order makes sense.

Reply:
looks ok now.